users/mixins.py

```diff
@@ -0,0 +1,17 @@
+from django.contrib import messages
+from django.contrib.auth.mixins import UserPassesTestMixin
+from django.urls import reverse
+
+
+class UserIsPartOfBattleMixin(UserPassesTestMixin):
+    permission_denied_message = 'You\'re not part of this battle!'
+    redirect_field_name = None
+
+    def test_func(self):
```

Comment:
This function should be renamed to something that express its purpose. Something like `check_user_is_creator_or_opponent` would do.

Reply:
It's part of UserPassesTestMixin, [check here](https://ccbv.co.uk/projects/Django/2.0/django.contrib.auth.mixins/UserPassesTestMixin/#test_func). Would it maybe solve the problem if I put a inline comment?